### PR TITLE
Added AR Tag Capabilities

### PR DESCRIPTION
**Pull request by mmartin4972**
* Created new camera interface to break apart camera logic
* Added config file
* Added AR Tag processing
* Moved data sets to the perception-files [repository](https://github.com/umrover/perception-files)
---
* TODO:
* The only supported resolution right now is 320x180 because of how we are opening the camera
* Currently compiling with -g debug flag in CMake File

### CODE REVIEW

File: jetson/percep_obs_detect/src/camera/filesystem.cpp [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+}
+
+void FileSystem::ignore_grab() {
+    no_grab = true;
+}
+
+cv::Mat& FileSystem::get_image() {
+    string filename = read_location + "/" + RGB_FOLDER + "/" + 
+                    RGB_FOLDER + std::to_string(frame_counter) + ".jpg";
+    image = cv::imread(filename.c_str());
```
Comment (qhdwight) [inline jetson/percep_obs_detect/src/camera/filesystem.cpp]: c_str() not needed but that's minor

File: jetson/percep_obs_detect/src/ar_tag/artag_detector.cpp [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+    #ifndef NO_JARVIS
+    lcm::LCM lcm_;
+    #endif
+    rover_msgs::TargetList arTagsMessage;
+    rover_msgs::Target* arTags = arTagsMessage.targetList;
+    arTags[0].distance = DEFAULT_TAG_VAL;
+    arTags[1].distance = DEFAULT_TAG_VAL;
+    
+    Mat rgb;
+    Mat src = cam->get_image();
```
Comment (qhdwight) [inline jetson/percep_obs_detect/src/ar_tag/artag_detector.cpp]: I'm not exactly sure how semantics work for the cv Mat type, but could this be reference?

**Review by qhdwight — COMMENTED** [review]
Looks really good fr fr but I would personally autoformat all the files.

**Review by arschallwig — APPROVED** [review]
Looks good

**Comment by mmartin4972** [comment]
> Looks really good fr fr but I would personally autoformat all the files.

What do you do to autoformat c++ files? Have never done that before.

**Comment by mmartin4972** [comment]
I made most of the changes you requested @qhdwight. I can't test them today. Will test them when I get access to the Perception laptop

File: jetson/percep_obs_detect/src/ar_tag/artag_detector.cpp (hunk 2 above)
Comment (mmartin4972) [reply jetson/percep_obs_detect/src/ar_tag/artag_detector.cpp]: Changed, but untested

File: jetson/percep_obs_detect/src/camera/filesystem.cpp (hunk 1 above)
Comment (mmartin4972) [reply jetson/percep_obs_detect/src/camera/filesystem.cpp:51]: Didn't want to mess w this one

File: jetson/percep_obs_detect/include/obstacle/obs-detector.h [hunk 3]
```diff
@@ -101,33 +94,28 @@ class ObsDetector {
...
+        
         // Lcm
         #ifndef NO_JARVIS
         lcm::LCM lcm_;
         rover_msgs::Obstacle obstacleMessage;
         #endif
 
         // Data sources
-        sl::Camera zed;
+        Source::Camera* cam;
```
[the remark this answers is not in the document]
Comment (mmartin4972) [reply jetson/percep_obs_detect/include/obstacle/obs-detector.h]: Changed, but untested

File: jetson/percep_obs_detect/CMakeLists.txt [hunk 4]
```diff
@@ -64,5 +81,7 @@ else ()
     target_compile_options(jetson_percep_obs_detect PRIVATE $<$<COMPILE_LANGUAGE:CUDA>:
             #-res-usage
             --use_fast_math
-            >)
+            >
+            -g # TODOD: REMOVE WHEN done debugging
```
[the remark this answers is not in the document]
Comment (mmartin4972) [reply jetson/percep_obs_detect/CMakeLists.txt]: Changed, but untested

**Comment by qhdwight** [comment]
Tested, made a new commit to fix some minors things

**Comment by mmartin4972** [comment]
Added the fixes you guys made and ran the auto-formatting

**Comment by mmartin4972** [comment]
> Oh lord that was all source files, okay I think I meant only your changes but I suppose this is fine too. Hopefully not too many merge conflicts in the future...

@niwhsa9 @qhdwight  do you guys think I should revert the auto formatting changes? I just figured it would be weird if half the code conformed to one style and the other half to another code style

**Comment by niwhsa9** [comment]
is it compiling like this? If so its fine by me.

**Comment by qhdwight** [comment]
Moved to #895

**Pull request closed without merging**